test/AbstractHandle_server_test.py

```diff
@@ -168,10 +169,36 @@ def test_persist_handle_ok(self):
         with self.assertRaises(ValueError) as context:
             handler.persist_handle(self.ctx, new_handle)[0]
 
-        self.assertIn('Connot insert doc', str(context.exception.args))
+        self.assertIn('Please do not specify hid', str(context.exception.args))
 
         self.mongo_util.delete_one(handle)
 
+    def test_persist_multiple_handles_ok(self):
+        self.start_test()
+        handler = self.getImpl()
+
+        handle = {'id': 'id',
+                  'file_name': 'file_name',
+                  'type': 'shock',
+                  'url': 'http://ci.kbase.us:7044/'}
+
+        counter = self.mongo_util.get_hid_counter()
+
+        thread_count = 257
+
+        threads = list()
+        for index in range(thread_count):
+            x = threading.Thread(target=handler.persist_handle, args=(self.ctx, handle))
+            threads.append(x)
+            x.start()
+
+        for index, thread in enumerate(threads):
+            thread.join()
+
+        new_counter = self.mongo_util.get_hid_counter()
+
+        self.assertEqual(counter + thread_count, new_counter)
```

Comment:
It'd be good to verify that all the returned handles have unique IDs.

Reply:
sounds good.